src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.PendingBranchesCollection.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class AbstractFlowPass<TLocalState, TLocalFunctionState>
+    {
+        protected sealed class PendingBranchesCollection : IEnumerable<PendingBranch>
+        {
+            private ArrayBuilder<PendingBranch> _branches;
+            private PooledDictionary<LabelSymbol, ArrayBuilder<PendingBranch>>? _branchesToLabel;
+
+            internal PendingBranchesCollection()
+            {
+                _branches = ArrayBuilder<PendingBranch>.GetInstance();
+            }
+
+            internal void Free()
+            {
+                _branches.Free();
+                FreeBranchesToLabel();
+            }
+
+            internal void Clear()
+            {
+                _branches.Clear();
+                FreeBranchesToLabel();
+            }
+
+            private void FreeBranchesToLabel()
+            {
+                if (_branchesToLabel is { })
+                {
+                    foreach (var branches in _branchesToLabel.Values)
+                    {
+                        branches.Free();
+                    }
+                    _branchesToLabel.Free();
+                    _branchesToLabel = null;
```

Comment:
Why do we `null` out `_branchesToLabel` but not other fields in `Free` operations?